cg/meta/demultiplex/utils.py

```diff
@@ -35,27 +35,35 @@ def get_lane_from_sample_fastq(sample_fastq_path: Path) -> int:
     raise ValueError(f"Could not extract lane number from fastq file name {sample_fastq_path.name}")
 
 
-def get_sample_id_from_sample_fastq(sample_fastq: Path) -> str:
-    """
-    Extract sample id from fastq file path.
+def get_sample_fastqs(flow_cell_directory: Path, pattern: str) -> List[Path]:
+    """Search for all files in a directory that match a pattern."""
+    return list(flow_cell_directory.glob(pattern))
 
-    Pre-condition:
-        - The parent directory name contains the sample id formatted as Sample_<sample_id>
-    """
 
-    sample_parent_match = re.search(r"Sample_(\w+)", sample_fastq.parent.name)
+def get_sample_fastqs_from_flow_cell(
+    flow_cell_directory: Path, sample_id: str
+) -> Optional[List[Path]]:
+    """Retrieve sample FastQs from a flow cell directory."""
+    root_pattern = f"{sample_id}_S*_L*_R*_*{FileExtensions.FASTQ}{FileExtensions.GZIP}"
+    unaligned_pattern = (
+        f"Unaligned*/Project_*/Sample_{sample_id}/*{FileExtensions.FASTQ}{FileExtensions.GZIP}"
+    )
 
-    if sample_parent_match:
-        return sample_parent_match.group(1)
-    else:
-        raise ValueError("Parent directory name does not contain 'Sample_<sample_id>'.")
+    for pattern in [root_pattern, unaligned_pattern]:
+        sample_fastqs = get_sample_fastqs(flow_cell_directory, pattern)
```

Comment:
```suggestion
        fastq_file_path: List[Path] = get_fastq_file_paths(flow_cell_directory, pattern)
```

argument names and type hint